tests/components/skybell/conftest.py

```diff
@@ -0,0 +1,177 @@
+    entry = MockConfigEntry(domain=DOMAIN, unique_id=USER_ID, data=CONF_DATA)
+    entry.add_to_hass(hass)
+    return entry
+
+
+@pytest.fixture()
+def config_entry(hass: HomeAssistant) -> MockConfigEntry:
+    """Add config entry in Home Assistant."""
+    return create_entry(hass)
+
+
+def patch_skybell_devices() -> None:
+    """Patch Skybell devices."""
+    mocked_skybell = AsyncMock()
+    mocked_skybell.user_id = USER_ID
+    return patch(
+        "homeassistant.components.skybell.config_flow.Skybell.async_get_devices",
+        return_value=[mocked_skybell],
+    )
+
+
+def patch_skybell() -> None:
+    """Patch Skybell."""
+    return patch(
+        "homeassistant.components.skybell.config_flow.Skybell.async_send_request",
+        return_value={"id": USER_ID},
+    )
+
+
+def patch_cache() -> None:
+    """Patch Skybell cache."""
+    return patch("aioskybell.utils.async_save_cache")
+
+
+async def set_aioclient_responses(aioclient_mock: AiohttpClientMocker) -> None:
+    """Set AioClient responses."""
+    aioclient_mock.get(
+        f"{BASE_URL}devices/{DEVICE_ID}/info/",
+        text=load_fixture("skybell/device_info.json"),
+    )
+    aioclient_mock.get(
+        f"{BASE_URL}devices/{DEVICE_ID}/settings/",
+        text=load_fixture("skybell/device_settings.json"),
+    )
+    aioclient_mock.get(
+        f"{BASE_URL}devices/{DEVICE_ID}/activities/",
+        text=load_fixture("skybell/activities.json"),
+    )
+    aioclient_mock.get(
+        f"{BASE_URL}devices/",
+        text=load_fixture("skybell/device.json"),
+    )
+    aioclient_mock.get(
+        USERS_ME_URL,
+        text=load_fixture("skybell/me.json"),
+    )
+    aioclient_mock.post(
+        f"{BASE_URL}login/",
+        text=load_fixture("skybell/login.json"),
+    )
+    aioclient_mock.get(
+        f"{BASE_URL}devices/{DEVICE_ID}/activities/1234567890ab1234567890ac/video/",
+        text=load_fixture("skybell/video.json"),
+    )
+    aioclient_mock.get(
+        f"{BASE_URL}devices/{DEVICE_ID}/avatar/",
+        text=load_fixture("skybell/avatar.json"),
+    )
+    aioclient_mock.get(
+        f"https://v3-production-devices-avatar.s3.us-west-2.amazonaws.com/{DEVICE_ID}.jpg",
+    )
+    aioclient_mock.get(
+        f"https://skybell-thumbnails-stage.s3.amazonaws.com/{DEVICE_ID}/1646859244793-951{DEVICE_ID}_{DEVICE_ID}.jpeg",
+    )
+
+
+@pytest.fixture
+async def connection(aioclient_mock: AiohttpClientMocker) -> None:
+    """Fixture for good connection responses."""
+    await set_aioclient_responses(aioclient_mock)
+
+
+async def async_init_integration(
+    hass: HomeAssistant,
+    skip_setup: bool = False,
+) -> MockConfigEntry:
+    """Set up the Skybell integration in Home Assistant."""
+    config_entry = create_entry(hass)
+
+    if not skip_setup:
+        with patch_cache(), patch(
+            "homeassistant.core.Config.path",
+            return_value="tests/components/skybell/fixtures/cache.pickle",
+        ):
```

Comment:
I admit this is not ideal. Any suggestions on how to divert the cache? Cache is included in tests. The tests run slow or require heavy patching to run fast.

Reply:
Maybe have a single test case that runs slow, and keep the test as you have it.  That seems a good balance between including everything but not slowing things down too much.